### PR TITLE
Add `address` as parameter to Connection constructor

### PR DESCRIPTION
**Pull request by vogdb**
Use `Helpers.extend` for all passing of `options`.

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review

File: test/plugin.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ describe('Plugin tests', function() {
     var plugin;
 
     beforeEach(function() {
-      plugin = new Plugin(new Session(new Connection, 'id'), 'name', 'id');
+      plugin = new Plugin(new Session(new Connection('id', {address: 'address'}), 'id'), 'name', 'id');
```
Comment (tomaszdurka) [inline test/plugin.js]: I did not pay too much attention when reviewing specs. But you should not need to adjust session/plugin specs when changing constructor of Connection. You should use mocks for Session/Connection.

File: src/connection.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ var Session = require('./session');
  * @extends TTransactionGateway
  */
 function Connection(id, options) {
+  Connection.validateOptions(options);
```
Comment (tomaszdurka) [inline src/connection.js]: Hmmm, what is the reason we have options in the first place? Not simple address?

File: test/plugin.js (hunk 1 above)
Comment (vogdb) [inline test/plugin.js]: Well. A simple mock `sinon.createStubInstance(Connection)` won't work. We need a more complex mock:
```js
var connectionMock = sinon.createStubInstance(Connection);
connectionMock.getOptions.returns({});
```
Cause this form is longer to write than `new Connection('id', {address: 'address'})`, I used the latter. Please confirm once again that I should use `connectionMock`

File: src/connection.js (hunk 2 above)
Comment (vogdb) [inline src/connection.js]: What about docu from above?
```js
/**
 * @param {string} id
 * @param {Object} options
 * @param {string} options.address
 * @param {string} [options.token]
 * @param {string} [options.apisecret]
 * @param {boolean|number} [options.keepalive]
 *
 * Important! Please listen to `error` events on a newly created instance in Node environment. For more details please look https://nodejs.org/api/events.html#events_error_events.
 * @constructor
 * @extends TEventEmitter
 * @extends TTransactionGateway
 */
function Connection(id, options) {
```
?

File: test/plugin.js (hunk 1 above)
Comment (vogdb) [inline test/plugin.js]: @tomaszdurka ?

Comment (tomaszdurka) [inline test/plugin.js]: Well mocking methods is always the same work to do, plus you don't need to adjust anything unless the mocked method changes. This is crucial to not over-adjust when changing tests. Given you *mock* similar way in multiple tests, constructor changes will bring loads of unrelated specs changes.

Just something to consider as it's just different (imo worse only in very short run) point of view.

File: src/connection.js (hunk 2 above)
Comment (tomaszdurka) [inline src/connection.js]: Basically I vote for using options for params which are... optional :) Great for token, apisecret, but in case of address which is mandatory I would keep separate param.

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by tomaszdurka** [comment]
lgtm
